using all the threads with smt leads to >60% of the cpu time being stuck in
kernel contention and doing nothing.

seems to be only a musl issue, though.
--- a/build/config/compiler/BUILD.gn
+++ a/build/config/compiler/BUILD.gn
@@ -678,7 +678,7 @@
       # and let it use all cores.
       # TODO(thakis): Check if '=0' (that is, number of cores, instead
       # of "all" which means number of hardware threads) is faster.
-      ldflags += [ "-Wl,--thinlto-jobs=all" ]
+      ldflags += [ "-Wl,--thinlto-jobs=0" ]
       if (is_apple) {
         ldflags +=
             [ "-Wl,-cache_path_lto," +
